src/ocean/text/convert/Integer.d

```diff
@@ -30,6 +30,133 @@ import ocean.transition;
 
 import ocean.core.Traits;
 
+import ocean.core.array.Search : find;
+
+import ocean.util.ReusableException;
+
+import ocean.math.Math;
+
+
+/*******************************************************************************
+
+    Parses a floating point number represented as a string directly to an
+    integer value.
+
+    To represent the fractional part we multiply the value by the requested
+    amount of decimal points and add it up. For example:
+
+    "1.123" -> 1123  (decimal_points = 3)
+    "0.01"  ->   10  (decimal_points = 3)
+
+    Any characters longer than the requested amount of decimal points will be
+    cut off:
+
+    "1.2345"  ->  123 (decimal_points = 2)
+    "10.2030" -> 1020 (decimal_points = 2)
+
+    Params:
+        decimal_points = amount of decimal points to consider
+        T              = type of the integer
+        value          = floating point number string to parse
+        exception      = optional, a reusable exception instance to throw
+
+    Returns:
+        the resulting integer number
+
+*******************************************************************************/
+
+public T floatStringToInt ( size_t decimal_points, T = ulong )
+                          ( cstring value, ReusableException exception = null )
+{
+
+    char[decimal_points] zeros_suffix_buf = '0';
+    const T multiplier = pow(cast(T)10, decimal_points);
+    cstring[2] num_parts;
+    ulong number = 0;
+
+    // Split string at '.'
+    auto idx = find(value, '.');
+
+    if (idx == value.length)
+    {
+        num_parts[0] = value;
+        num_parts[1] = zeros_suffix_buf;
+    }
+    else
+    {
+        num_parts[0] = value[0 .. idx];
+        num_parts[1] = value[idx+1..$];
+    }
+
+    // Cut off if too long
+    if (num_parts[1].length > decimal_points)
+        num_parts[1].length = decimal_points;
+
+    // Fill with zeros if too short
+    if (num_parts[1].length < decimal_points)
+    {
+        zeros_suffix_buf[0 .. num_parts[1].length] = num_parts[1];
+        num_parts[1] = zeros_suffix_buf;
+    }
+
+    ulong tryParse ( cstring str )
```

Comment:
At least, I saw a warning once when I ran `-vgc` with D2 and it complained about these things..